cdap-ui/app/cdap/components/FieldLevelLineage/v2/LineageSummary/index.tsx

```diff
@@ -28,40 +30,151 @@ const styles = (theme) => {
+    const sourceEl = d3.select(`#${source}`);
+    const destEl = d3.select(`#${destination}`);
+
+    const offsetX = -100; // From the padding on the LineageSummary
+    const offsetY = -50; // From the FllHeader
+
+    const sourceXY = sourceEl.node().getBoundingClientRect();
+    const destXY = destEl.node().getBoundingClientRect();
+
+    const sourceX1 = sourceXY.right + offsetX;
+    const sourceY1 = sourceXY.top + offsetY + 0.5 * sourceXY.height;
+    const sourceX2 = destXY.left + offsetX;
+    const sourceY2 = destXY.top + offsetY + 0.5 * sourceXY.height;
+
+    // draw an edge from line start to line end
+    const linkContainer = d3.select(`#${source}_${destination}`);
+
+    const third = (sourceX2 - sourceX1) / 3;
+
+    // Draw a line with a bit of curve between the straight parts
+    const lineGenerator = d3.line().curve(d3.curveMonotoneX);
+
+    const points = [
+      [sourceX1, sourceY1],
+      [sourceX1 + third, sourceY1],
+      [sourceX2 - third, sourceY2],
+      [sourceX2, sourceY2],
+    ];
+
+    linkContainer
+      .append('path')
+      .style('stroke', '#bbbbbb')
+      .style('stroke-width', '1')
+      .style('fill', 'none')
+      .attr('d', lineGenerator(points));
+
+    // draw left anchor
+    const anchorHeight = 8;
+    const anchorRx = 1.8;
+    linkContainer
+      .append('rect')
+      .attr('x', sourceX1 - anchorHeight * 0.5)
+      .attr('y', sourceY1 - anchorHeight * 0.5)
+      .attr('width', anchorHeight)
+      .attr('height', anchorHeight)
+      .attr('rx', anchorRx)
+      .style('fill', '#bbbbbb');
+
+    // draw right anchor
+    linkContainer
+      .append('rect')
+      .attr('x', sourceX2 - anchorHeight * 0.5)
+      .attr('y', sourceY2 - anchorHeight * 0.5)
+      .attr('width', anchorHeight)
+      .attr('height', anchorHeight)
+      .attr('rx', anchorRx)
+      .style('fill', '#bbbbbb');
+  }
+
+  public drawLinks() {
+    // clear any existing links and anchors
+    d3.select('#links-container')
+      .selectAll('path,rect')
+      .remove();
+
+    this.activeLinks.forEach((link) => {
+      this.drawLineFromLink(link);
+    });
+  }
+
+  public componentWillUnmount() {
+    (document.body as HTMLBodyElement).onresize = null;
+  }
+
+  public componentDidMount() {
+    this.drawLinks();
+    (document.body as HTMLBodyElement).onresize = debounce(this.drawLinks.bind(this), 1);
+  }
+  public render() {
+    return (
+      <Consumer>
+        {({
+          causeSets,
+          target,
+          targetFields,
+          impactSets,
+          firstCause,
+          firstImpact,
+          firstField,
+          links,
+        }) => {
+          this.activeLinks = links;
+          return (
+            <div className={this.props.classes.root} id="fll-container">
+              <svg id="links-container" className={this.props.classes.container}>
+                <g>
+                  {links.map((link) => {
+                    const id = `${link.source}_${link.destination}`;
```

Comment:
Could this `id` potentially clash?
```
source: `one`
target: `two_three`
```
and
```
source: `one_two`
target: `three`
```

Reply:
I've updated all table and field id's to prevent this clash (as well as clashes from cycles in lineage).